crypto/cms/cms_lib.c

```diff
@@ -551,11 +549,9 @@ int CMS_add0_cert(CMS_ContentInfo *cms, X509 *cert)
 
 int CMS_add1_cert(CMS_ContentInfo *cms, X509 *cert)
 {
-    int r;
-    r = CMS_add0_cert(cms, cert);
-    if (r > 0)
-        X509_up_ref(cert);
-    return r;
+    int r = CMS_add0_cert(cms, cert);
+
+    return r > 0 ? X509_up_ref(cert) : r;
```

Comment:
However, here you would have to up_ref first and then free() the reference on failure.

Reply:
Done